source/dotnet/Application/Batches/BatchApplicationService.cs

```diff
@@ -81,6 +81,10 @@ public async Task UpdateExecutionStateAsync()
                 batch.Complete();
                 completedBatches.Add(batch);
             }
+            else if (state == JobState.Canceled)
+            {
+                batch.Restart();
```

Comment:
Perhaps `batch.Complete()` is a bad name. But I was a little surprised to see a restart method on the aggregate. I would expect to find a restart on a service.

```suggestion
                batch.ResetStatus();
```

Reply:
I agree, the naming is a bit artificial. I have renamed to ResetStatus, MarkAsCompleted and MarkAsExecuting.

The design is maybe too simplistic at the moment. I expect we are going to need to refactor JobRunId into a list of (JobRunId, State, Version) on a batch, because we are going to need to track re-runs of the same batch. A batch may then loose these methods as state will be tracked per run.